src/python/pants/engine/rules.py

```diff
@@ -317,14 +329,26 @@ def _ensure_type_annotation(
   return annotation
 
 
-def rule(*args, cacheable=True) -> Callable:
+PUBLIC_RULE_DECORATOR_ARGUMENTS = set(['name'])
+
+
+def rule_decorator(*args, **kwargs) -> Callable:
   if len(args) != 1 and not inspect.isfunction(args[0]):
     raise ValueError(
       'The @rule decorator expects no arguments and for the function it decorates to be '
       f'type-annotated. Given {args}.'
     )
 
+  if len(set(kwargs) - PUBLIC_RULE_DECORATOR_ARGUMENTS - set(['cacheable'])) != 0:
```

Comment:
Shouldn't `cacheable` be added to the `PUBLIC_DECORATOR_ARGUMENTS` set?

Reply:
I think it deliberately shouldn't be. The API pants already exposes for controlling whether a rule is cacheable or not is the distinction between `@rule` and `@console_rule`, so we don't want to tell people that they can use `cacheable` as a kwarg. In fact we might even want to write some additional code that would throw an error if rule code tried to manually set `cacheable=False` rather than relying on the `@console_rule` decorator.

Of course this is assuming that we want to keep the distinction between `@rule` and `@console_rule` long-term, but I think we've decided that we do want to keep this distinction.